app/src/main/java/com/hedvig/app/feature/home/ui/claimstatus/composables/Pills.kt

```diff
@@ -0,0 +1,103 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.intl.Locale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.hedvig.app.feature.home.ui.claimstatus.data.PillData
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import com.hedvig.app.ui.compose.theme.hedvigContentColorFor
+
+@Composable
+fun Pills(pillData: List<PillData>) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.Top,
+    ) {
+        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
+            pillData.forEach { pillData: PillData ->
+                Pill(
+                    text = pillData.text,
+                    pillType = pillData.type,
+                )
+            }
+        }
+        // Uncomment when the card is clickable to go to detail screen
+//        Icon(Icons.Default.ArrowForward, contentDescription = null)
+    }
+}
+
+@Composable
+private fun Pill(
+    text: String,
+    pillType: PillData.PillType
+) {
+    val backgroundColor = when (pillType) {
+        is PillData.PillType.Contained -> pillType.color
+        PillData.PillType.Outlined -> Color.Transparent
+    }
+    Surface(
+        shape = MaterialTheme.shapes.small,
+        color = backgroundColor,
+        contentColor = when (pillType) {
+            is PillData.PillType.Contained -> hedvigContentColorFor(backgroundColor)
+            PillData.PillType.Outlined -> hedvigContentColorFor(MaterialTheme.colors.background)
+        },
+        border = when (pillType) {
+            is PillData.PillType.Contained -> null
+            PillData.PillType.Outlined -> BorderStroke(
+                width = ButtonDefaults.OutlinedBorderSize,
+                color = MaterialTheme.colors.onSurface
+            )
+        }
+    ) {
+        Row(
+            Modifier.padding(
+                horizontal = 8.dp,
+                vertical = 5.dp
+            ),
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            // TODO uppercase it in a different way? Check user's preference on language maybe?
+            val currentLocale = Locale.current
+            Text(
+                text.uppercase(java.util.Locale(currentLocale.language, currentLocale.region)),
+                style = MaterialTheme.typography.caption,
+                maxLines = 1
+            )
```

Comment:
Is there a better way to force uppercase in this scenario? We do have limited language options anyway, so would looking at the user selected language be the best approach? Is there some other way that I am not aware of?

Reply:
This is probably the right way to do it, but shouldn't we have been provided with this text already pre-uppercased? 🤔